cli/formatter/format.go

```diff
@@ -0,0 +1,800 @@
+package formatter
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"sort"
+	"strconv"
+	"strings"
+	"unicode/utf8"
+
+	"github.com/apex/log"
+	"github.com/jedib0t/go-pretty/v6/table"
+	"github.com/jedib0t/go-pretty/v6/text"
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	defaultFormatStr     = ""
+	yamlFormatStr        = "yaml"
+	yamlFormatShortStr   = "y"
+	luaFormatStr         = "lua"
+	luaFormatShortStr    = "l"
+	tableFormatStr       = "table"
+	tableFormatShortStr  = "t"
+	ttableFormatStr      = "ttable"
+	ttableFormatShortStr = "T"
+)
+
+// Format defines a set of supported output format.
+type Format int
+
+const (
+	DefaultFormat Format = iota
+	YamlFormat
+	LuaFormat
+	TableFormat
+	TTableFormat
+)
+
+// ParseFormat parses a output format string representation. It supports
+// mixed case letters.
+func ParseFormat(str string) (Format, bool) {
+	switch str {
+	case defaultFormatStr:
+		return DefaultFormat, true
+	case yamlFormatStr, yamlFormatShortStr:
+		return YamlFormat, true
+	case luaFormatStr, luaFormatShortStr:
+		return LuaFormat, true
+	case tableFormatStr, tableFormatShortStr:
+		return TableFormat, true
+	case ttableFormatStr, ttableFormatShortStr:
+		return TTableFormat, true
+	}
+	return DefaultFormat, false
+}
+
+// String returns a string representation of the output format.
+func (f Format) String() string {
+	switch f {
+	case DefaultFormat:
+		return defaultFormatStr
+	case YamlFormat:
+		return yamlFormatStr
+	case LuaFormat:
+		return luaFormatStr
+	case TableFormat:
+		return tableFormatStr
+	case TTableFormat:
+		return ttableFormatStr
+	default:
+		panic("Unknown output format")
+	}
+}
+
+// MakeOutput returns formatted output depending on specified output format and formatting options.
+func MakeOutput(data string, outputFormat Format, opts *Opts) string {
```

Comment:
The `MakeOutput` function does not change opts.

```suggestion
func MakeOutput(data string, outputFormat Format, opts Opts) string {
```

Reply:
Done